src/clients/InventoryClient.ts

```diff
@@ -300,80 +298,87 @@ export class InventoryClient {
-    } else {
-      this.logger.debug({
-        at: "InventoryClient#getAllBundleRefunds",
-        message: "Refunds from upcoming bundle",
-        refunds: nextBundleRefunds[0],
-      });
+  // Return sum of refunds for all fills sent after the fromBlocks.
+  // Makes a simple assumption that all fills sent by this relayer that were sent after the last executed bundle
+  // are valid and will be refunded on the repayment chain selected.
+  private _getApproximateRefundsForToken(
+    l1Token: EvmAddress,
+    fromBlocks: { [chainId: number]: number }
+  ): { [repaymentChainId: number]: BigNumber } {
+    const refundsForChain: { [repaymentChainId: number]: BigNumber } = {};
+    for (const _chainId of Object.keys(fromBlocks)) {
+      const chainId = Number(_chainId);
+      const spokePoolClient = this.tokenClient.spokePoolClients[chainId];
+      if (!isDefined(spokePoolClient)) {
+        continue;
+      }
+      spokePoolClient
+        .getFills()
+        .filter(
+          (fill) =>
+            compareAddressesSimple(fill.relayer.toNative(), this.relayer.toNative()) &&
+            // We can assume that this.getL1TokenAddress will succeed here because this function's input variable l1Token
+            // is also retrieved via the same function.
+            compareAddressesSimple(
+              l1Token.toNative(),
+              this.getL1TokenAddress(fill.inputToken, fill.originChainId).toNative()
+            ) &&
+            fill.blockNumber >= fromBlocks[chainId]
+        )
+        .forEach((fill) => {
+          const { inputAmount: refundAmount, repaymentChainId } = fill;
+
+          const existingRefundAmount = refundsForChain[repaymentChainId] ?? bnZero;
+          refundsForChain[repaymentChainId] = existingRefundAmount.add(refundAmount);
+        });
     }
-
-    mark.stop({
-      message: "Time to calculate total refunds per chain",
-    });
-    return refunds;
+    return refundsForChain;
   }
 
-  async executeBundleRefundsPromise(): Promise<CombinedRefunds[]> {
-    if (!isDefined(this.bundleRefundsPromise)) {
-      this.bundleRefundsPromise = this.getAllBundleRefunds();
-    }
-    return this.bundleRefundsPromise;
-  }
+  // Return the next starting block for each chain following the bundle end block of the last exececuted bundle that
+  // was relayed to that chain.
+  private _getUpcomingRefundsQueryFromBlocks(): { [chainId: number]: number } {
+    const configStoreClient = this.hubPoolClient.configStoreClient;
+    return Object.fromEntries(
+      this.chainIdList.map((chainId) => {
+        const spokePoolClient = this.tokenClient.spokePoolClients[chainId];
+        // Step 1: Find the last RelayedRootBundle event that was relayed to this chain. Assume this contains refunds
+        // from the last executed bundle for this chain and these refunds were executed.
+        const lastRelayedRootToChain =
+          spokePoolClient.getRootBundleRelays()[spokePoolClient.getRootBundleRelays().length - 1];
```

Comment:
Should be able to use `Array.at()` here to avoid calling `spokePoolClient.getRootBundleRelays()` twice.

```suggestion
          spokePoolClient.getRootBundleRelays().at(-1);
```